src/service/programService.js

```diff
@@ -1228,51 +1131,112 @@ function programList(req, response) {
-        },
-        attributes: data.request.fields || {
-          include : [[Sequelize.json('config.subject'), 'subject'], [Sequelize.json('config.defaultContributeOrgReview'), 'defaultContributeOrgReview'], [Sequelize.json('config.framework'), 'framework'], [Sequelize.json('config.board'), 'board'],[Sequelize.json('config.gradeLevel'), 'gradeLevel'], [Sequelize.json('config.medium'), 'medium']],
-          exclude: ['config', 'description']
-        },
-        offset: res_offset,
-        limit: res_limit,
-        order: [
-          ['updatedon', 'DESC']
-        ]
+    }
+  }
+  else {
+    if (data.request.filters && data.request.filters.role && data.request.filters.user_id) {
+      const promises = [];
+      const roles = data.request.filters.role;
+      const user_id = data.request.filters.user_id;
+      delete data.request.filters.role;
+      delete data.request.filters.user_id;
+
+      _.forEach(roles, (role) => {
+          let whereCond = {
+            $contains: Sequelize.literal(`cast(rolemapping->>'${role}' as text) like ('%${user_id}%')`),
+          };
+
+          promises.push(
+            model.program.findAndCountAll({
+            where: {
+              ...whereCond,
+              ...data.request.filters,
+              ...filters
+            },
+            offset: res_offset,
+            limit: res_limit,
+            order: [
+              ['updatedon', 'DESC']
+            ]
+        })
+        )
       })
+      Promise.all(promises)
       .then(function (res) {
-        let apiRes = _.map(res, 'dataValues');
-        if (data.request.sort){
-          apiRes = programServiceHelper.sortPrograms(apiRes, data.request.sort);
-        }
+        let aggregatedRes = [];
+        _.forEach(res, (response) => {
+          _.forEach(response.rows, row => aggregatedRes.push(row));
+        })
+        aggregatedRes = _.uniqBy(aggregatedRes, 'dataValues.program_id');
         loggerService.exitLog({responseCode: 'OK'}, logObject);
         return response.status(200).send(successResponse({
           apiId: 'api.program.list',
           ver: '1.0',
           msgid: uuid(),
           responseCode: 'OK',
           result: {
-            count: apiRes ? apiRes.length : 0,
-            programs: apiRes || []
+            count: aggregatedRes.length,
+            programs: aggregatedRes
           }
         }))
       })
       .catch(function (err) {
+        console.log(err)
         loggerService.exitLog({responseCode: 'ERR_LIST_PROGRAM'}, logObject);
-        loggerError('',rspObj,errCode+errorCodes.CODE5);
+        loggerError('',rspObj,errCode+errorCodes.CODE4);
         return response.status(400).send(errorResponse({
           apiId: 'api.program.list',
           ver: '1.0',
           msgid: uuid(),
           responseCode: 'ERR_LIST_PROGRAM',
           result: err
-        },errCode+errorCodes.CODE5));
+        },errCode+errorCodes.CODE4));
       });
+    } else {
+      model.program.findAll({
```

Comment:
let us use `await` and `try catch` to handle promises. we won't end up in nested functions.

Reply:
Fixed